### PR TITLE
feat(aws): ECR Image Scan Findings

### PR DESCRIPTION
**Pull request by bbernays**

### CODE REVIEW

**Review by erezrokah — APPROVED** [review]
🚀

**Review by hermanschaaf — APPROVED** [review]
LGTM; I can't really think of a better way to write the resolver either. Just left a few nits / questions - maybe I'm forgetting something about how the codegen unwrapping code works.

File: plugins/source/aws/codegen/recipes/ecr.go [hunk 1]
```diff
@@ -83,6 +84,15 @@ func ECRResources() []*Resource {
 						Options:  schema.ColumnCreationOptions{PrimaryKey: true},
 					},
 				}...),
+			Relations: []string{"RepositoryImageScanFindings()"},
+		},
+		{
+			SubService:            "repository_image_scan_findings",
+			Struct:                &models.ImageScanWrapper{},
+			SkipFields:            []string{"RepositoryArn"},
```
[the remark this answers is not in the document]
Comment (disq) [reply plugins/source/aws/codegen/recipes/ecr.go:92]: Not on any level, looks like a copy-paste artifact.

File: plugins/source/aws/resources/services/ecr/repositories_mock_test.go [hunk 2]
```diff
@@ -41,6 +41,24 @@ func buildEcrRepositoriesMock(t *testing.T, ctrl *gomock.Controller) client.Serv
 	}
 	m.EXPECT().ListTagsForResource(gomock.Any(), gomock.Any(), gomock.Any()).Return(&tagResponse, nil)
 
+	iF := types.ImageScanFindings{}
```
Comment (disq) [inline plugins/source/aws/resources/services/ecr/repositories_mock_test.go:44]: why not fake a `DescribeImageScanFindingsOutput` and set `NextToken` to `nil`?

File: plugins/source/aws/codegen/recipes/ecr.go [hunk 3]
```diff
@@ -83,6 +84,15 @@ func ECRResources() []*Resource {
...
+			UnwrapEmbeddedStructs: false,
```
[the remark this answers is not in the document]
Comment (bbernays) [reply plugins/source/aws/codegen/recipes/ecr.go]: This was a relic of an earlier version

File: plugins/source/aws/codegen/recipes/ecr.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (bbernays) [reply plugins/source/aws/codegen/recipes/ecr.go]: Copy and paste error

**Pull request merged**